src/Orleans/AzureUtils/StatsTableDataManager.cs

```diff
@@ -147,16 +147,16 @@ private StatsTableData PopulateStatsTableDataEntry(ICounter statsCounter)
             counter++;
             var time = DateTime.UtcNow;
             var timeStr = time.ToString(DATE_FORMAT, CultureInfo.InvariantCulture);
-            entry.PartitionKey = deploymentId + ":" + timeStr;
+
+            // number of ticks remaining until the year 9683
+            var ticks = DateTime.MaxValue.Ticks - time.Ticks;
+
+            // partition the table according to the deployment id and hour
+            entry.PartitionKey = string.Join("$", deploymentId, string.Format("{0:d19}", ticks - ticks % TimeSpan.TicksPerHour));
```

Comment:
Should we pad this with a leading zero to make it always be of even length? See also at http://gauravmantri.com/2012/02/17/effective-way-of-fetching-diagnostics-data-from-windows-azure-diagnostics-table-hint-use-partitionkey/ wherein padding is used, albeit one could think with a reference to authority.